src/SFA.DAS.PAS.Infrastructure.UnitTests/Services/TrainingProviderApiClientTest.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.PAS.Infrastructure.UnitTests.Services
+{
+    public class TrainingProviderApiClientTest
+    {
+        private const string OuterApiBaseAddress = "http://outer-api";
+        private MockHttpMessageHandler _mockHttp = null!;
+        private Fixture _fixture = null!;
+        private TrainingProviderApiClient _sut = null!;
+        private TrainingProviderApiClientConfiguration _config = null!;
+        private Mock<ILogger<TrainingProviderApiClient>> _logger = null!;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _fixture = new Fixture();
+            _mockHttp = new MockHttpMessageHandler();
+            _config = _fixture
+                .Build<TrainingProviderApiClientConfiguration>()
+                .With(x =>x.ApiBaseUrl, OuterApiBaseAddress)
+                .With(x => x.IdentifierUri, "")
+                .Create<TrainingProviderApiClientConfiguration>();
+            _logger = new Mock<ILogger<TrainingProviderApiClient>>();
+
+            var client = new HttpClient(_mockHttp)
+            {
+                BaseAddress = new Uri(OuterApiBaseAddress),
+            };
+            _sut = new TrainingProviderApiClient(client, _config, _logger.Object);
+        }
+
+        [Test]
+        public async Task When_ProviderDetails_Found_Then_Data_FromOuterApiIsReturned()
+        {
+            // Arrange
+            var ukprn = _fixture.Create<long>();
+            var expected = _fixture.Create<GetProviderSummaryResult>();
+
+            _mockHttp.When($"{OuterApiBaseAddress}/api/providers/{ukprn}")
+                .Respond("application/json", 
+                    JsonSerializer.Serialize(expected));
+
+            // Act
+            var actual = await _sut.GetProviderDetails(ukprn);
+
+            // Assert
+            actual.Should().BeEquivalentTo(expected);
+        }
+
+        [Test]
+        public async Task When_ProviderDetails_NotFound_Then_Data_FromOuterApiIsNotFound()
+        {
+            // Arrange
+            var ukprn = _fixture.Create<long>();
+            _mockHttp.When($"{OuterApiBaseAddress}/api/providers/{ukprn}")
+                .Respond("application/json", 
+                    JsonSerializer.Serialize((GetProviderSummaryResult)null));
+
+            // Act
+            var actual = await _sut.GetProviderDetails(ukprn);
+
+            // Assert
+            actual.Should().BeNull();
+        }
+
+        [Test]
+        public async Task When_ProviderDetails_InternalServerError_Then_Data_FromOuterApiIsNotFound()
+        {
+            // Arrange
+            var ukprn = _fixture.Create<long>();
+
+            var error = @"{""Code"":""ServiceUnavailable"", ""Message"" : ""Service Unavailable.""}";
+
+            _mockHttp.When(HttpMethod.Get, $"{OuterApiBaseAddress}/api/providers/{ukprn}")
```

Comment:
for the above and this can do
```
var response = new HttpResponseMessage
            {
                Content = new StringContent("\"test\""),
                StatusCode = HttpStatusCode.InternalServerError
            };
var httpMessageHandler = new Mock<HttpMessageHandler>();
            httpMessageHandler.Protected()
                .Setup<Task<HttpResponseMessage>>(
                    "SendAsync",
                    ItExpr.Is<HttpRequestMessage>(c =>
                        c.Method.Equals(method)
                        && c.RequestUri.AbsoluteUri.Equals(baseUrl)),
                    ItExpr.IsAny<CancellationToken>()
                )
                .ReturnsAsync((HttpRequestMessage request, CancellationToken token) => response);
            return httpMessageHandler;
```

Reply:
Removed the third party Nuget Pkg and replaced the test cases with native mock http handler.